text/0000-crates.io-default-ranking.md

```diff
@@ -0,0 +1,1440 @@
+
+### Overall
+
+Since documentation/ease of use was such a highly mentioned factor in peoples' decisions, we propose that, instead of averagaing the three scores, we weight ease of use by 2x and divide by 4 instead of 3.
+
+<table>
+    <tr>
+        <th>Crate</th>
+        <th>Ease of use</th>
+        <th>Maintenance</th>
+        <th>Popularity</th>
+        <th>Overall</th>
+    </tr>
+    <tr>
+        <td>nom</td>
+        <td>92%, ☀️</td>
+        <td>78%, ⛅️</td>
+        <td>100%, ☀️</td>
+        <td>91%, ☀️</td>
+    </tr>
+    <tr>
+        <td>combine</td>
+        <td>103%, ☀️</td>
+        <td>68%, 🌥</td>
+        <td>60%, 🌥</td>
+        <td>84%, 🌤</td>
+    </tr>
+    <tr>
+        <td>peg</td>
+        <td>66%, 🌥</td>
+        <td>100%, ☀️</td>
+        <td>80%, 🌤</td>
+        <td>78%, ⛅️</td>
+    </tr>
+    <tr>
+        <td>lalrpop</td>
+        <td>80%, 🌤</td>
+        <td>53%, ☁️</td>
+        <td>40%, 🌧</td>
+        <td>63%, 🌥</td>
+    </tr>
+    <tr>
+        <td>peresil</td>
+        <td>105%, ☀️</td>
+        <td>3%, 🌧</td>
+        <td>20%, 🌧</td>
+        <td>58%, ☁️</td>
+    </tr>
+</table>
+
+
+## Display
+
+On a list of crates, the letter representing the score in each category plus
+the overall score would be displayed using text, color, symbols, and detail on
+hover, with a link to a more thorough explanation. We like the information
+density in the way [npms][] displays scores:
+
+<img src="http://i.imgur.com/yadRNyy.png" src="Example npms score circles" width="442" />
+
+## Out of scope
+
+This proposal is not advocating to change the order of **search results**; those
+should still be ordered by relevancy to the query based on the indexed content.
+We may want to have an option to sort search results by "recommended" or
+whatever we want to call this sorting, but probably not change the default.
+
+# How do we teach this?
+
+A criticism we anticipate and that would be totally fair is that this formula
+is too complex. If we go with this formula, we think it's important to make
+available a clear explanation of why a crate has the score it does, for
+transparency to both crate users and crate authors. [Ruby toolbox][ruby] has a
+great example of what we'd like to provide.
+
+[ruby]: #ruby-toolbox
+
+A possible benefit of having multiple measures influence the ranking is making
+it less likely that crate owners will go to the effort of gaming the formula in
+order to have a higher ranking.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+We might create a system that incentivizes attributes that are not useful, or
+worse, actively harmful to the Rust ecosystem. For example, the documentation
+percentage could be gamed by having one line of uninformative documentation for
+all public items, thus giving a score of 100% without the value that would come
+with a fully documented library. We hope the community at large will agree
+these attributes are valuable to approach in good faith, and that trying to
```

Comment:
I think the Rust community in particular will do well with a "good faith" mechanism.